Kudu.Core/Infrastructure/SecurityUtility.cs

```diff
@@ -1,21 +1,44 @@
-﻿using System;
+﻿using Microsoft.AspNetCore.DataProtection;
+using Microsoft.Azure.Web.DataProtection;
+using System;
 using System.Security.Cryptography;
 
 namespace Kudu.Core.Infrastructure
 {
     public static class SecurityUtility
     {
-        public static string GenerateSecretString()
+        private const string DefaultProtectorPurpose = "function-secrets";
+
+        private static string GenerateSecretString()
         {
             using (var rng = RandomNumberGenerator.Create())
             {
                 byte[] data = new byte[40];
                 rng.GetBytes(data);
                 string secret = Convert.ToBase64String(data);
-
                 // Replace pluses as they are problematic as URL values
                 return secret.Replace('+', 'a');
             }
         }
+
+        public static Tuple<string, string>[] GenerateSecretStringsKeyPair(int number)
+        {
+            var unencryptedToEncryptedKeyPair = new Tuple<string, string>[number];
+            var protector = DataProtectionProvider.CreateAzureDataProtector().CreateProtector(DefaultProtectorPurpose);
+            for (int i = 0; i < number; i++)
+            {
+                string unencryptedKey = GenerateSecretString();
+                unencryptedToEncryptedKeyPair[i] = new Tuple<string, string>(unencryptedKey, protector.Protect(unencryptedKey));
+            }
+            return unencryptedToEncryptedKeyPair;
+        }
+
+        public static string DecryptSecretString(string content)
```

Comment:
it is called [here](https://github.com/projectkudu/kudu/pull/2157/files/f43631a94f7ae08732bffc0c2efaf6acde95d98a#diff-770a930cac2c5747cde5d2d9940eeca2R245)